### PR TITLE
test: deflaking settings test more

### PR DESCRIPTION
**Pull request by alyssawilk**
Signed-off-by: Alyssa Wilk <alyssar@chromium.org>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: test/integration/multiplexed_integration_test.cc [hunk 1]
```diff
@@ -1788,7 +1788,10 @@ TEST_P(Http2FrameIntegrationTest, AdjustUpstreamSettingsMaxStreams) {
   ASSERT_TRUE(fake_upstreams_[0]->waitForRawConnection(fake_upstream_connection1));
   const Http2Frame settings_frame = Http2Frame::makeSettingsFrame(
       Http2Frame::SettingsFlags::None, {{NGHTTP2_SETTINGS_MAX_CONCURRENT_STREAMS, 1}});
-  ASSERT_TRUE(fake_upstream_connection1->write(std::string(settings_frame)));
+  std::string settings_data(settings_frame);
+  ASSERT_TRUE(fake_upstream_connection1->write(settings_data));
+  test_server_->waitForCounterGe("cluster.cluster_0.upstream_cx_rx_bytes_total",
+                                                                  settings_data.size());
```
Comment (wrowe) [inline test/integration/multiplexed_integration_test.cc:1794]: https://dev.azure.com/cncf/envoy/_build/results?buildId=106994&view=logs&j=55bc2ab9-36c1-5ab8-58a6-8c258728eeff&t=33b57bf9-ef13-58c8-bc2b-f804aeb16d39&l=805

**Review by wrowe — CHANGES_REQUESTED** [review]
Small format quirk to repair

**Comment by alyssawilk** [comment]
PTAL!

**Pull request merged**